pkg/cmd/version.go

```diff
@@ -0,0 +1,209 @@
+`
+)
+
+type VersionOptions struct {
+	printFlags  *genericclioptions.PrintFlags
+	configFlags *genericclioptions.ConfigFlags
+
+	restConfig      *rest.Config
+	kubeClient      kubernetes.Interface
+	discoveryClient discovery.CachedDiscoveryInterface
+	dynamicClient   dynamic.Interface
+
+	podUrlGetter *util.RemotePodURLGetter
+
+	fileWriter *util.MultiSourceFileWriter
+	builder    *resource.Builder
+	args       []string
+
+	// directory where all gathered data will be stored
+	baseDir string
+	// whether or not to allow writes to an existing and populated base directory
+	overwrite bool
+
+	genericclioptions.IOStreams
+}
+
+func NewVersionOptions(streams genericclioptions.IOStreams) *VersionOptions {
+	return &VersionOptions{
+		printFlags:  genericclioptions.NewPrintFlags("gathered").WithDefaultOutput("yaml").WithTypeSetter(scheme.Scheme),
+		configFlags: genericclioptions.NewConfigFlags(),
+		IOStreams:   streams,
+	}
+}
+
+func NewCmdVersion(parentName string, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewVersionOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:          "version [flags]",
+		Short:        "Gather debugging data for a given Cluser Version",
+		Example:      fmt.Sprintf(versionExample, parentName),
+		SilenceUsage: true,
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := o.Complete(c, args); err != nil {
+				return err
+			}
+			if err := o.Validate(); err != nil {
+				return err
+			}
+			if err := o.Run(); err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVar(&o.baseDir, "base-dir", "must-gather", "Root directory used for storing all gathered cluster operator data. Defaults to $(PWD)/must-gather")
+	cmd.Flags().BoolVar(&o.overwrite, "overwrite", false, "If true, allow this command to write to an existing location with previous data present")
+
+	o.printFlags.AddFlags(cmd)
+	return cmd
+}
+
+func (o *VersionOptions) Complete(cmd *cobra.Command, args []string) error {
+	o.args = args
+
+	var err error
+	o.restConfig, err = o.configFlags.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+
+	o.kubeClient, err = kubernetes.NewForConfig(o.restConfig)
+	if err != nil {
+		return err
+	}
+
+	o.dynamicClient, err = dynamic.NewForConfig(o.restConfig)
+	if err != nil {
+		return err
+	}
+
+	o.discoveryClient, err = o.configFlags.ToDiscoveryClient()
+	if err != nil {
+		return err
+	}
+
+	printer, err := o.printFlags.ToPrinter()
+	if err != nil {
+		return err
+	}
+	o.fileWriter = util.NewMultiSourceWriter(printer)
+	o.podUrlGetter = &util.RemotePodURLGetter{
+		Protocol: "https",
+		Host:     "localhost",
+		Port:     "8443",
+	}
+
+	// pre-fetch token while we perform other tasks
+	if err := o.podUrlGetter.FetchToken(o.restConfig); err != nil {
+		return err
+	}
+
+	o.builder = resource.NewBuilder(o.configFlags)
+	return nil
+}
+
+func (o *VersionOptions) Validate() error {
+	if len(o.args) >= 1 {
+		return fmt.Errorf("this option takes not arguments")
```

Comment:
`len(o.args) > 0` reads a bit better imo. Also, how about rephrasing to `this command does not take arguments`